salt/state.py

```diff
@@ -53,26 +57,24 @@
 from salt.template import compile_template, compile_template_str
 from salt.utils.odict import DefaultOrderedDict, OrderedDict
 
+
 log = logging.getLogger(__name__)
 
+# See https://docs.saltproject.io/en/latest/ref/states/layers.html for details on the naming
+# __exclude__ and __extend__ are list values; the other items are state items are
+# state items that have dict values. This could be cleaned up some by making
+# exclude and extend into dicts instead of lists so the type of all the values are
+# homogenous.
```

Comment:
```suggestion
# See https://docs.saltproject.io/en/latest/ref/states/layers.html for details on the naming
# __exclude__ and __extend__ are list values; the other items are state items 
# that have dict values. This could be cleaned up some by making
# exclude and extend into dicts instead of lists so the type of all the values are
# homogeneous.
```